server/accounts.go

```diff
@@ -2329,11 +2360,17 @@ func (a *Account) checkStreamImportAuthorizedNoLock(account *Account, subject st
 	return a.checkStreamExportApproved(account, subject, imClaim)
 }
 
-func (a *Account) checkAuth(ea *exportAuth, account *Account, imClaim *jwt.Import) bool {
+func (a *Account) checkAuth(ea *exportAuth, account *Account, imClaim *jwt.Import, tokens []string) bool {
 	// if ea is nil or ea.approved is nil, that denotes a public export
-	if ea == nil || (ea.approved == nil && !ea.tokenReq) {
+	if ea == nil || (ea.approved == nil && !ea.tokenReq && ea.accountPos == 0) {
 		return true
 	}
+	// Check if the export is protected and enforces presence of importing account identity
+	if ea.accountPos > 0 && ea.accountPos <= uint(len(tokens)) {
+		if tokens[ea.accountPos-1] == account.Name {
+			return true
+		}
```

Comment:
Is it possible that `ea.approved == nil` and `ea.tokenReq == false` and `ea.accountPos != 0` but not found in tokens? Code will not panic because Go does not when doing `m[key]` on a nil map, but is that not risky? Should we return false here?

Reply:
Agreed, will return false